consensus/replay.go

```diff
@@ -252,15 +252,10 @@ func (h *Handshaker) Handshake(proxyApp proxy.AppConns) (string, error) {
 	}
 	appHash := res.LastBlockAppHash
 
-	h.logger.Info("ABCI Handshake App Info",
-		"height", blockHeight,
-		"hash", appHash,
-		"software-version", res.Version,
-		"protocol-version", res.AppVersion,
-	)
-
-	// Only set the version if there is no existing state.
-	if h.initialState.LastBlockHeight == 0 {
+	appVersion := h.initialState.Version.Consensus.App
```

Comment:
> // Only set the version if there is no existing state.

do you know the reasoning for this comment? seems not right?

Reply:
like, the comment just seems like a flaw in logic, I'm curious tho if anyone know more context around it

Reply:
> do you know the reasoning for this comment? seems not right?

It just means if we're starting from zero we take the appVersion returned from `Info` else if we're restarting a node and are at a non-zero height we should already have the latest appVersion.

To be fair in that case, it's probably worth checking that they are in sync